cli/exp_task_send_test.go

```diff
@@ -171,3 +181,65 @@ func Test_TaskSend(t *testing.T) {
 		})
 	}
 }
+
+func Test_TaskSend_HappyPath(t *testing.T) {
+	t.Parallel()
+
+	client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+	owner := coderdtest.CreateFirstUser(t, client)
+	userClient, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID)
+	ctx := testutil.Context(t, testutil.WaitLong)
+
+	fakeAPI := startFakeAgentAPI(t, map[string]http.HandlerFunc{
+		"/status": func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			_ = json.NewEncoder(w).Encode(map[string]string{
+				"status": "stable",
+			})
+		},
+		"/message": func(w http.ResponseWriter, r *http.Request) {
+			// The agentapi SDK expects a Message response
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			var msg aiagentapi.PostMessageParams
+			if err := json.NewDecoder(r.Body).Decode(&msg); err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+			require.Equal(t, "Please add unit tests", msg.Content)
+			message := aiagentapi.Message{
+				Id:      999,
+				Role:    aiagentapi.RoleAgent,
+				Content: "You got it",
+				Time:    time.Now(),
+			}
+			_ = json.NewEncoder(w).Encode(message)
+		},
+	})
+	authToken := uuid.NewString()
+	template := createAITaskTemplate(t, client, owner.OrganizationID, withSidebarURL(fakeAPI.URL()), withAgentToken(authToken))
+
+	wantPrompt := "build me a calculator"
+	workspace := coderdtest.CreateWorkspace(t, userClient, template.ID, func(req *codersdk.CreateWorkspaceRequest) {
+		req.RichParameterValues = []codersdk.WorkspaceBuildParameter{
+			{Name: codersdk.AITaskPromptParameterName, Value: wantPrompt},
+		}
+	})
+	coderdtest.AwaitWorkspaceBuildJobCompleted(t, client, workspace.LatestBuild.ID)
+
+	agentClient := agentsdk.New(client.URL, agentsdk.WithFixedToken(authToken))
+	_ = agenttest.New(t, client.URL, authToken, func(o *agent.Options) {
+		o.Client = agentClient
+	})
+
+	coderdtest.NewWorkspaceAgentWaiter(t, client, workspace.ID).
+		WaitFor(coderdtest.AgentsReady)
+
+	var stdout strings.Builder
+	inv, root := clitest.New(t, "exp", "task", "send", workspace.Name, "Please add unit tests")
+	inv.Stdout = &stdout
+	clitest.SetupConfig(t, userClient, root)
+
+	err := inv.WithContext(ctx).Run()
+	require.NoError(t, err)
+}
```

Comment:
How about we just merge this with `Test_TaskSend`? I think having duplication, and manually defined endpoints increases the maintenance burden and if we're starting coderd anyway, we can just dbgen the necessary fixtures and use those for the table tests.

If we need a way to modify the responses coderd returns for some tests, that could be achieved by setting `coderdtest.Options.APIMiddleware` for pre/post processing.

Reply:
So if I understand correctly, you're suggesting that we migrate the existing table tests to use `coderdtest`? 

Also, TIL about `coderdtest.Options.APIMiddleware` 👍

Reply:
Yep, that's correct 👍🏻